apps/backend/src/stripe/index.ts

```diff
@@ -148,30 +148,58 @@ const checkSubscriptionPaymentMethodFilled = async (
   return customer.invoice_settings.default_payment_method !== null;
 };
 
-const getArgosSubscriptionDataFromStripeSubscription = async (
-  subscription: Stripe.Subscription,
+async function getIncludedScreenshotsFromStripeSubscription(
+  stripeSubscription: Stripe.Subscription,
+) {
+  const firstItem = getFirstItemFromStripeSubscription(stripeSubscription);
+  const price = firstItem.price;
+  if (price.billing_scheme !== "tiered") {
+    return null;
+  }
+  const { tiers } = await stripe.prices.retrieve(price.id, {
+    expand: ["tiers"],
+  });
+  invariant(tiers, "no tiers found in a tiered price");
+
+  // Take flat tiers only
+  const flatTiers = tiers.filter((tier) => tier.up_to && tier.flat_amount);
+  // Take the last one (the highest number of screenshots allowed)
```

Comment:
Are the tiers sorted by size ? The comment says "the highest" but you picked the last one. You should use a `Math.max()` instead of last index.